### PR TITLE
fix: get rid of deployment config in servlet context initializer

### PR DESCRIPTION
**Pull request by denis-anisimov**
* Implement a Spring application config factory
* Use the Spring config app factory in Spring boot case
* Adapt unit tests to use the proper config

### CODE REVIEW

**Review by caalador — COMMENTED** [review]
Seems fine to me. Soroosh can have a look also as I had not that much time.

File: vaadin-spring/src/test/java/com/vaadin/flow/spring/SpringStubServletConfigTest.java [hunk 1]
```diff
@@ -1,84 +0,0 @@
...
-        when(registration.getServletName()).thenReturn("vaadinServlet");
-        when(context.getInitParameterNames())
-                .thenReturn(Collections.emptyEnumeration());
-        when(registration.getInitParameters())
-                .thenReturn(Collections.emptyMap());
-
-    }
-
-    @Test
-    public void vaadinProductionMode_isReadFromEnvironmentVariables() {
```
Comment (pleku) [inline vaadin-spring/src/test/java/com/vaadin/flow/spring/SpringStubServletConfigTest.java:64]: Non-blocking question: Would this have made any sense anymore as another test without the stub config ?

**Review by pleku — DISMISSED** [review]
Just one question, otherwise LGTM

File: vaadin-spring/src/test/java/com/vaadin/flow/spring/SpringStubServletConfigTest.java (hunk 1 above)
Comment (pleku) [reply vaadin-spring/src/test/java/com/vaadin/flow/spring/SpringStubServletConfigTest.java:64]: If not, good riddance. If yes, then can be added back later (to get this in)

Comment (denis-anisimov) [reply vaadin-spring/src/test/java/com/vaadin/flow/spring/SpringStubServletConfigTest.java:64]: There is already a test which replaces this one.
Except this one checks only one single property `productionMode`.
`SpringApplicationConfigurationFactoryTest`  tests all `SpringServlet.PROPERTY_NAMES` properties.
And `SpringApplicationConfigurationFactory` is the proper SPI impl replacement for ugly stub config

**Review by pleku — APPROVED** [review]

**Pull request merged**